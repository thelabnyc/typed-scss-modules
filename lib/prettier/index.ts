import prettier from "prettier";
import { alerts } from "../core/index.ts";


/**
 * Format input using prettier
 *
 * @param {file} file
 * @param {string} input
 */
export const applyPrettier = async (file: string, input: string) => {
  try {
    const config = await prettier.resolveConfig(file, {
      editorconfig: true,
    });
    // try to return formatted output
    return prettier.format(input, { ...config, parser: "typescript" });
  } catch (error) {
    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
    alerts.notice(`Tried using prettier, but failed with error: ${error}`);
  }

  // failed to format
  return input;
};
