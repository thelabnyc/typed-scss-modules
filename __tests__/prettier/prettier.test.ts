import {jest} from "@jest/globals";
import path from "node:path";
import prettier from "prettier";
import { applyPrettier } from "../../lib/prettier/index.ts";
import { classNamesToTypeDefinitions } from "../../lib/typescript/index.ts";

import { fileURLToPath } from "node:url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const file = path.join(__dirname, "test.d.ts");
const input =
  "export type Styles = {'myClass': string;'yourClass': string;}; export type Classes = keyof Styles; declare const styles: Styles; export default styles;";

describe("applyPrettier", () => {
  it("should locate and apply prettier.format", async () => {
    const output = await applyPrettier(file, input);

    expect(await prettier.format(input, { parser: "typescript" })).toMatch(output);
  });

  it("should match snapshot", async () => {
    const typeDefinition = await classNamesToTypeDefinitions({
      banner: "",
      classNames: ["nestedAnother", "nestedClass", "someStyles"],
      file,
      exportType: "default",
    });

    if (!typeDefinition) {
      throw new Error("failed to collect typeDefinition");
    }

    const output = await applyPrettier(file, typeDefinition);

    expect(output).toMatchSnapshot();
  });
});

