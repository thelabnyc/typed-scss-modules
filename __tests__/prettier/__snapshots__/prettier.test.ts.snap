// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`applyPrettier should match snapshot 1`] = `
"export type Styles = {
    nestedAnother: string;
    nestedClass: string;
    someStyles: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
"
`;
